### DISCUSSION TITLE
How to emulate Zephyr Timers for Test-Driven Development of state machines

### DISCUSSION THREAD

**Discussion by vChavezB** [zephyrproject-rtos/zephyr#56582] [Q&A] [answered]
Hello Zephyr community,

I am starting to delve in to Test-Driven Development (TDD) and I have developed c++ state machines with the [QP CPP framework](https://github.com/QuantumLeaps/qpcpp). I would like to use ZTEST to unit test my state machines but my code depends on the zephyr timer API.

In my tests I would like to go fast forward in time (emulation) so I can test if my state machines are triggered and check the functional behavior without having to wait. 

Would it be possible to link my own timer implementation for the timer APIs (e.g. `k_timer_start`) for a posix board and how could I implement this ? are there any examples that emulate the timer API?

**Comment by yashi** [comment, marked as answer]
Try [`CONFIG_NATIVE_POSIX_SLOWDOWN_TO_REAL_TIME=n`](https://docs.zephyrproject.org/latest/kconfig.html#CONFIG_NATIVE_POSIX_SLOWDOWN_TO_REAL_TIME) on either `native_posix` or `native_posix_64`.

>  If deselected, the process will run as fast as possible.

**Reply by vChavezB** [reply]
Thanks for the info, I will check it out :)
